### ISSUE TITLE
Bug: `_get_weights_and_dims` doesn't work properly

### ISSUE THREAD

**Issue by andersy005** [NCAR/esmlab#91]
Does anybody have a straightforward example of using the `dim` keyword on `weighted_corr`? I'm having issues with it. I'm testing the p-value function on, say, a lat/lon grid of data.

```python
import numpy as np
import xarray as xr
import esmlab

base = np.random.rand(100,10,10)
x = xr.DataArray(np.random.rand(100,10,10) + base, dims=['time','lat','lon'])
y = xr.DataArray(np.random.rand(100,10,10) + base, dims=['time','lat','lon'])
# just looking to correlate over the 'time' dimension so a grid of correlation
# coefficients is returned.
r = esmlab.statistics.weighted_corr(x, y, dim=['time'], weights=None) 
```

<img width="920" alt="Screen Shot 2019-03-09 at 9 03 48 PM" src="https://user-images.githubusercontent.com/8881170/54080519-dbbf4980-42ae-11e9-9a72-8b939ca3c88a.png">

_Originally posted by @bradyrx in https://github.com/NCAR/esmlab/pull/90#issuecomment-471245176_

**Comment by andersy005** [comment]
It appears that  the issue is in`_apply_nan_mask()`. This function assumes that **all dimensions** should be used:
https://github.com/NCAR/esmlab/blob/479fa7093f2991a5c58ae9bd0ad5fd4724981ef3/esmlab/statistics.py#L51-L53

https://github.com/NCAR/esmlab/blob/479fa7093f2991a5c58ae9bd0ad5fd4724981ef3/esmlab/statistics.py#L21-L32

As a consequence, the shape of the returned `weights` is always equal to the input array's shape, even when the user wants to apply the operation along a particular dimension. 

To fix this, we will need to come up with a way of doing `valid = x.notnull() ` while taking into consideration  dimensions supplied by the user.

**Comment by andersy005** [comment]
Fixed in #97
